package.json

```diff
@@ -233,15 +233,13 @@
           ],
           "description": "Have VSCode download the language server automatically (MacOS/Windows only for Swift-MesonLSP)"
         },
-        "mesonbuild.setupCppTools": {
-          "type": "boolean",
-          "default": true,
-          "markdownDescription": "Setup the [Microsoft C/C++ extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) to work with Meson"
-        },
-        "mesonbuild.setupRustAnalyzer": {
-          "type": "boolean",
+        "mesonbuild.modifySettings": {
+          "type": [
+            "boolean",
+            "array"
+          ],
           "default": true,
-          "markdownDescription": "Setup the [rust-analyzer extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) to work with Meson"
+          "markdownDescription": "Automatically setup other extensions to use files generated by Meson. Disable this if for example .vscode/settings.json is committed into git and should not be modified. Can also be an array of settings that can be modified, currently [cppTools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) or [rustAnalyser](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools)."
```

Comment:
```
"Automatically setup other extensions to use files generated by Meson. Disable this if, for example, `.vscode/settings.json` is tracked by version control and should not be modified. It can also be an array of extension IDs of the following:\\n- [`ms-vscode.cpptools`](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools)\\n -[`rust-lang.rust-analyzer`](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer)."
```

Reply:
done